src/main/scala/com/github/unchama/generic/ListExtra.scala

```diff
@@ -75,4 +77,21 @@ object ListExtra {
     map(firstTarget.product(secondTarget))
   }
 
+  /**
+   * 最初の要素が `predicate` を満たすようになるまで `list` を左回転する。
+   *
+   * `list` が空である場合、または `list` を一周回転させても最初の要素が
+   * `predicate` を満たすことが無かった (つまり、すべての要素が
+   * `predicate` を満たさなかった) 場合には `None` が返される。
+   */
+  @tailrec
+  def rotateLeftUntil[A](list: List[A])(predicate: A => Boolean): Option[List[A]] = {
+    list.find(predicate) match {
+      case Some(_) =>
+        if (predicate(list.head)) Some(list)
+        else rotateLeftUntil(list.filterNot(_ == list.head) :+ list.head)(predicate)
+      case _ => None
+    }
+  }
```

Comment:
```suggestion
  def rotateLeftUntil[A](list: List[A])(predicate: A => Boolean): Option[List[A]] = {
    val rotationSize = list.indexWhere(predicate)
    if (rotationSize == -1) {
      None
    } else {
      val (cutPrefix, cutSuffix) = list.splitAt(rotationSize)
      Some(cutSuffix ++ cutPrefix)
    }
  }
```

元の実装では head と重複している要素が消えてしまうので仕様を満たせていません。また、計算量的にも append を最大 list.length-1 回やるのは重いので、 `indexWhere + splitAt` にしました。